### ISSUE TITLE
User typo: $$ is meant to be &&

### ISSUE THREAD

**Issue by TinCanTech** [koalaman/shellcheck#1446]
#### For bugs
- Rule Id (if any, e.g. SC1000): n/a
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0 + Online
- [ X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ X ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ X ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
foo="temp"
rm -r "$foo" $$ echo "removed $foo" || echo "failed to remove $foo"
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

This is _obviously_ a typo in my script ($$ is meant to be &&) and  my own fault.  But as _shellcheck finds bugs in your shell scripts_, I was surprised it did not detect this. I understand, this may be beyond the scope of shellcheck, thus, only reporting this for your consideration. (If there is a previous case for this please direct me as I could not find one)

**Comment by koalaman** [comment]
I definitely see what you mean, and a human would certainly flag this. I wasn't able to find any other instances of this though, while there were several legitimate `rm foo$$` type things that used it legitimately as a temporary filenames. Since ShellCheck has a hard time detecting things that aren't systematic errors, I guess we can leave this as a mostly one-off occurrence. 

Thanks for the suggestion!
